pkg/gencosmosdb/cosmosdb/template_fake.go

```diff
@@ -0,0 +1,301 @@
+package cosmosdb
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/ugorji/go/codec"
+
+	pkg "github.com/jim-minter/go-cosmosdb/pkg/gencosmosdb/cosmosdb/dummy"
+)
+
+type FakeTemplateDecoder func(*string) *pkg.Template
+type FakeTemplateTrigger func(context.Context, *pkg.Template) error
+type FakeTemplateQuery func(*Query, map[string]*string, FakeTemplateDecoder) []*string
+
+var _ TemplateClient = &FakeTemplateClient{}
+
+type FakeTemplateClient struct {
+	docs       map[string]*string
+	jsonHandle *codec.JsonHandle
+	lock       sync.Locker
+	triggers   map[string]FakeTemplateTrigger
+	queries    map[string]FakeTemplateQuery
+}
+
+func NewFakeTemplateClient(h *codec.JsonHandle) *FakeTemplateClient {
+	return &FakeTemplateClient{
+		docs:       make(map[string]*string),
+		triggers:   make(map[string]FakeTemplateTrigger),
+		queries:    make(map[string]FakeTemplateQuery),
+		jsonHandle: h,
+		lock:       &sync.Mutex{},
+	}
+}
+
+func (c *FakeTemplateClient) fromString(s *string) *pkg.Template {
+	res := &pkg.Template{}
+	d := codec.NewDecoder(bytes.NewBufferString(*s), c.jsonHandle)
+	err := d.Decode(&res)
+	if err != nil {
+		panic(err)
```

Comment:
Sure. I was not sure if "running into unintended/impossible behaviour" should be a panic or bubbled-up error -- my concern is that it might pass unseen (since Queries don't/can't return errors directly? I guess we could return it in the Iterator...)

Reply:
Removed all the panics. Not clear what to do where there's no way of clearly bubbling things up, short of adding additional errors to the real interface where there aren't already.